File: Problem001/erlang/problem.erl
-module(problem).
-export([main/0]).

main()->
  Num=numList(999),
  Sum=getSum(Num),
  io:format("~b\n",[Sum]).

numList(0)->[];
numList(N)->[N|numList(N-1)].

getSum(List)->getSum(List,0).
getSum([],Acc)->Acc;
getSum(List,Acc) when hd(List) rem 3 == 0 orelse hd(List) rem 5==0 -> getSum(tl(List),Acc+hd(List));
getSum(List,Acc) -> getSum(tl(List),Acc).

File: Problem003/kotlin/problem.kt

fun main(args:Array<String>)
{
    val N:Long=600851475143
    println(getMaxFactor(N))
}
fun getMaxFactor(n:Long):Long
{
    var factor:Long=2
    var num=n
    while(num>1)
    {
        when (num % factor) {
            0L ->
            {
                num /=factor
            }
            else -> factor++
        }

    }
    return factor
}
